runtime/src/topo/hwloc/topo-hwloc.c

```diff
@@ -311,6 +312,9 @@ void getCPUInfo(void) {
     CHK_ERR_ERRNO(core = hwloc_get_ancestor_obj_by_type(topology,
                                                          HWLOC_OBJ_CORE,
                                                          pu));
+    int numPus = hwloc_bitmap_weight(core->cpuset);
+    CHK_ERR((pusPerCore == 0) || (pusPerCore == numPus));
+    pusPerCore = numPus;
```

Comment:
See the in-line comment below, but there isn't a good way to get the number of cores from the online cpuset because PUs and cores that are online but inaccessible don't have representative objects in the topology. So you can't call `hwloc_get_nbobjs_inside_cpuset_by_type(...,HWLOC_OBJ_CORE...)` for example. I make the assumption that all cores have the same number of PUs and compute the number of cores from the numbers of PUs.